yellowbrick/features/pca.py

```diff
@@ -226,111 +281,128 @@ def transform(self, X, y=None, **kwargs):
             Returns a new array-like object of transformed features of shape
             ``(len(X), proj_dim)``.
         """
-        self.pca_features_ = self.pca_transformer.transform(X)
-        self.draw()
-        return self.pca_features_
-
-    def draw(self, **kwargs):
+        try:
+            Xp = self.pca_transformer.transform(X)
+            self.draw(Xp, y)
+            return Xp
+        except NotFittedError:
+            raise NotFitted.from_estimator(self, "transform")
+
+    def draw(self, Xp, y):
         """
         Plots a scatterplot of points that represented the decomposition,
         `pca_features_`, of the original features, `X`, projected into either 2 or
         3 dimensions.
 
         If 2 dimensions are selected, a colorbar and heatmap can also be optionally
         included to show the magnitude of each feature value to the component.
+        
+        Parameters
+        ----------
+        Xp : array-like of shape (n, 2) or (n, 3)
+            The matrix produced by the ``transform()`` method.
+
+        y : array-like of shape (n,), optional
+            The target, used to specify the colors of the points.
+
 
         Returns
         -------
-        self : visualizer.ax
-            Returns the axes of the visualizer for use in Pipelines
+        self.ax : matplotlib Axes object
+            Returns the axes that the scatter plot was drawn on.
         """
-        X = self.pca_features_
-        if self.proj_dim == 2:
-            im = self.ax.scatter(
-                X[:, 0],
-                X[:, 1],
-                c=self.color,
-                cmap=self.colormap,
-                alpha=self.alpha,
-                vmin=self.pca_components_.min(),
-                vmax=self.pca_components_.max(),
-                **kwargs
-            )
-            if self.colorbar:
+        # Call to super draw which draws the scatter plot.
+        super(PCADecomposition, self).draw(Xp, y)
+        if self.proj_features:
+            # Draws projection features in transformed space.
+            self.draw_projection_features(Xp, y)
+        if self.projection == 2:
+            if self.heatmap:
+                # TODO: change to pcolormesh instead of imshow per #615 spec
```

Comment:
We don't have to do this in this PR - but we should consider it for the next PR.

Reply:
Sure!